test/boost/cql_query_test.cc

```diff
@@ -125,6 +125,34 @@ SEASTAR_TEST_CASE(test_create_twcs_table_no_ttl) {
     });
 }
 
+SEASTAR_TEST_CASE(test_twcs_max_window) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        // Create a TWCS table with no defined TTL
+        e.execute_cql("CREATE TABLE tbl (a int, b int, PRIMARY KEY (a)) WITH "
+                      "compaction = {'class': 'TimeWindowCompactionStrategy', "
+                      "'compaction_window_size': '1', 'compaction_window_unit': 'HOURS'};").get();
+        e.require_table_exists("ks", "tbl").get();
+        // Creating a TWCS table with a large number of windows/buckets should fail
+        assert_that_failed(e.execute_cql("CREATE TABLE tbl2 (a int, b int, PRIMARY KEY (a)) WITH "
+                      "compaction = {'class': 'TimeWindowCompactionStrategy', "
+                      "'compaction_window_size': '1', 'compaction_window_unit': 'HOURS'} "
+                      "AND default_time_to_live=864000;"));
+        // However the creation of a table within bounds should succeed
+        e.execute_cql("CREATE TABLE tbl2 (a int, b int, PRIMARY KEY (a)) WITH "
+                      "compaction = {'class': 'TimeWindowCompactionStrategy', "
+                      "'compaction_window_size': '1', 'compaction_window_unit': 'HOURS'} "
+                      "AND default_time_to_live=86400;").get();
```

Comment:
This test depends on a specific setting of the default of the restriction (50). I wonder if the test shouldn't specify this restriction explicitly - otherwise if we change our mind in the future what the default should be, this test will break.
Or, on the other hand, maybe it's good to have this test to verify that the default can't change unintentionally?

Reply:
We now early LiveUpdate to ensure if we decide to change the default the tests won't break. ;-)